tests/security_test.go

```diff
@@ -0,0 +1,143 @@
+package tests
+
+import (
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func TestDefaultServiceAccounts(t *testing.T) {
+	const driverSA = "spark-service-account"
+	const operatorSA = "spark-operator-service-account"
+	spark := utils.SparkOperatorInstallation{
+		Namespace: "test-default-service-accounts",
+		Params: map[string]string{
+			"createOperatorServiceAccount": "true",
+			"createSparkServiceAccount":    "true",
+			"operatorServiceAccountName":   operatorSA,
+			"sparkServiceAccountName":      driverSA,
+		},
+	}
+
+	err := testServiceAccounts(spark,
+		utils.DefaultInstanceName+"-"+operatorSA,
+		utils.DefaultInstanceName+"-"+driverSA)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+}
+
+func TestSparkServiceAccount(t *testing.T) {
+	const namespace = "test-provided-spark-sa"
+	const driverSA = "spark-driver"
+	const operatorSA = "spark-operator-service-account"
+
+	// Prepare a namespace with a service account
+	err := createNamespaceAndServiceAccount(namespace, driverSA)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	spark := utils.SparkOperatorInstallation{
+		Namespace:            namespace,
+		SkipNamespaceCleanUp: true,
+		Params: map[string]string{
+			"createOperatorServiceAccount": "true",
+			"operatorServiceAccountName":   operatorSA,
+			"createSparkServiceAccount":    "false",
+			"sparkServiceAccountName":      driverSA,
+		},
+	}
+
+	err = testServiceAccounts(spark, utils.DefaultInstanceName+"-"+operatorSA, driverSA)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+}
+
+func TestOperatorServiceAccount(t *testing.T) {
+	const namespace = "test-provided-operator-sa"
+	const driverSA = "spark-service-account"
+	const operatorSA = "operator-test-service-account"
+
+	// Prepare a namespace with a service account
+	err := createNamespaceAndServiceAccount(namespace, operatorSA)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	spark := utils.SparkOperatorInstallation{
+		Namespace:            namespace,
+		SkipNamespaceCleanUp: true,
+		Params: map[string]string{
+			"createOperatorServiceAccount": "false",
+			"operatorServiceAccountName":   operatorSA,
+			"createSparkServiceAccount":    "true",
+			"sparkServiceAccountName":      driverSA,
```

Comment:
seems like a lot of this test code is duplicated? may I suggest a helper function that enables toggling b/w `true` `false` for the spark/operator service accounts... like you did with `testServiceAccounts`?